packages/atomic/cypress/integration/search-box.cypress.ts

```diff
@@ -102,11 +98,11 @@ describe('Search Box Test Suites', () => {
   });
 
   describe('search box with number of values set to 3', () => {
-    const htmlCode =
-      '<atomic-search-box number-of-suggestions="3"></atomic-search-box>';
     beforeEach(() => {
-      setUpPage(htmlCode);
-      generateAliasForSearchBox();
+      new TestFixture()
+        .with(addSearchBox({'number-of-suggestions': '3'}))
```

Comment:
I saw that adding an element is done with js, creating a DOM node and looping through the object passed here to set the attributes.

I'm wondering though if it would be possible to add a html string? I feel this would be a general api, avoiding the next to create helper functions like `addSearchBox` every time we add a component. The fixture could also handle inlining any js needed to execute a search (as it is already doing). e.g.

```
const page = new TestFixture()
.with('<atomic-search-box query-suggest="true"></atomic-search-box>')
.withFirstSearch(false)
.html();

const dom = render(page);

// tests
```


I'd say html is closer to how a user configures Atomic and maybe more readable, but beyond this, my question is: what is the benefit of [switching to use DOM nodes](https://github.com/coveo/ui-kit/pull/726/files#diff-28f32bf87b61cb6c20088527208ffdd60b987e5b6d21543f4039c7b4a5ce3bf9R124)?

Reply:
Yes, I can add a "inject raw HTML directly" function.

In the case of the searchbox specifically, I see that as an helper, because I'm guessing that many test will want to include a search box to test something meaningful.

Example: You want to test query summary by actually submitting a query from a search box, or did you mean etc.

So, for some components, I'm guessing we will want to include them/share them for many test. So it's easier to modify what should be the "default search box config" for all test that needs a search box.

We won't be creating helper for each and every single component.


As for dealing with DOM nodes versus raw HTML, I find it much easier to deal with, to remove/add prop.

Versus a string where we will quickly need to do some regex magic to configure the components.